### ISSUE TITLE
issue with creating and destroying CBS

### ISSUE THREAD

**Issue by likexian** [terraform-providers/terraform-provider-tencentcloud#82]
By @lubars:

Upgraded Terraform and Tencentcloud plugin:

> Terraform v0.12.5
> + provider.null v2.1.2
> + provider.tencentcloud v1.14.0

Still getting internal errors on roughly 25% of calls to _terraform apply_ (this succeeded on the third try):

> Error: [TencentCloudSDKError] Code=InternalError, Message=(bd6d22a359eb)An internal error has occurred. Retry your request, but if the problem persists, contact us with details by posting a message on the Tencent cloud forums., RequestId=5498c37c-ff95-4231-ab6c-bd6d22a359eb
> 
>   on infrastructure.tf line 372, in resource "tencentcloud_cbs_storage" "data":
>  372: resource "tencentcloud_cbs_storage" "data" {

Still seeing errors on calls to _terraform destroy_ as well:

> Error: [TencentCloudSDKError] Code=ResourceBusy, Message=656e6c70-be4a-4657-aed1-95964abc0b7b,ins-ruv3rbzm is busy, please retry later (0afe54a64aa6), RequestId=a7e786f4-5c73-47e6-8d37-0afe54a64aa6

from: https://github.com/terraform-providers/terraform-provider-tencentcloud/issues/18#issuecomment-516985527

**Comment by likexian** [comment]
Hello
please try the v1.14.1, if there is still some trouble, please open a new issue.

**Comment by lubars** [comment]
Now using:

> Terraform v0.12.6
> + provider.null v2.1.2
> + provider.tencentcloud v1.14.1

Allocation seems much improved, however I am unable to deallocate after 30+ attempts:

> Error: [TencentCloudSDKError] Code=RequestLimitExceeded, Message=您当前每秒请求 `21` 次，超过了每秒频率上限 `20`，请稍后重试。, RequestId=fcce6963-be7e-4bea-ad40-93e9e2c6af0f

**Comment by likexian** [comment]
Hello @lubars 
Thank you for your feedback, this failed due to api request rate limited, we will fix it.
By the way, how many disks did you Allocation/Deallocate once?

**Comment by lubars** [comment]
In this case, each "terraform apply" creates 4 x tencentcloud_instance, and each tencentcloud_instance has 1 x system disk + 4 x  tencentcloud_cbs_storage; I ran two of these "terraform apply" jobs in parallel (both jobs sharing a tencentcloud_vpc but otherwise distinct).

**Comment by likexian** [comment]
Got it, Thank you.

**Comment by lubars** [comment]
@likexian Please let me know when fixed (just tried with provider.tencentcloud v1.15.0 and behavior is unchanged), thanks.

**Comment by likexian** [comment]
@lubars No problem

**Comment by likexian** [comment]
Hello @lubars
We have fixed it, would you please upgrade to v1.15.1 and have a try?
Thank you.

**Comment by lubars** [comment]
Hi @likexian,

Version:
> Terraform v0.12.6
> + provider.null v2.1.2
> + provider.tencentcloud v1.15.1

I encountered multiple internal errors during the "terraform apply" phase:

> Error: tencentcloud_instance got error, code:TradeUnknownError, message:An internal error has occurred. Retry your request, but if the problem persists, contact us with details by posting a message on the Tencent cloud forums., request id:9ba01af9-dc14-4a3d-b671-6f06df2516ba

However, I am no longer seeing any errors during the "destroy" phase.

-Steve

**Comment by likexian** [comment]
Hello @lubars 
Sorry again for the trouble, I will fix this internal error in next release.

**Comment by likexian** [comment]
Hello @lubars
The issue of `internal errors` has been fixed, I think it will works well now.
Please have a try, if there is still something wrong, please reply to this thread.
Thank you very much.

**Comment by lubars** [comment]
Hi @likexian,

Updated to latest version:

```
# terraform version
Terraform v0.12.6
+ provider.null v2.1.2
+ provider.tencentcloud v1.15.2
```

Still seeing internal errors during "apply" phase:

> Error: tencentcloud_instance got error, code:TradeUnknownError, message:An internal error has occurred. Retry your request, but if the problem persists, contact us with details by posting a message on the Tencent cloud forums., request id:cf5f0019-a5b8-4609-a7d1-0275b726e738

Regards, @lubars

**Comment by likexian** [comment]
Hello @lubars
Sorry, we miss a bug and must be fixed in next version.

**Comment by lubars** [comment]
Hi @likexian, do you suggest I try v1.16.0?

**Comment by likexian** [comment]
Hello @lubars
Yes, please try it.

**Comment by lubars** [comment]
@likexian, behavior during apply/destroy appears much improved:  I did not encounter any errors in two runs of the test case I've been using (create two clusters in parallel, four compute instances per cluster, five disks per compute instance).

**Comment by likexian** [comment]
Hello @lubars , thank you very much for your feedback and patience with us.
I am happy that there is no error again, however if there is any problems in the future, please open a issue.
